### PR TITLE
Update release policy with our understanding of Breaking Change

### LINKED ISSUE #563: Establish policy about what is considered a breaking change on Solidus
[Semver](https://semver.org/) defines a clear policy for versioning new releases of software:

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
> - MAJOR version when you make incompatible API changes
> - MINOR version when you add functionality in a backwards compatible manner
> - PATCH version when you make backwards compatible bug fixes

However, it doesn’t define what a breaking change is.

[Solidus proclaims to adhere](https://solidus.io/release_policy/) to semver. However, we don’t specify which is our understanding of backward compatibility.

We want to define a BC in the following way:

A breaking change is any kind of change that can break an application using Solidus, excluding:

- Issues due to the application using an interface explicitly marked as private.
- Issues due to the application monkey patching Solidus code.
- Issues due to a fix for a security or very important bug for which it wasn’t possible to implement a non-breaking solution.
- Issues due to the application using a Ruby/Rails version that is no longer maintained by the Ruby/Rails team. In the same vein that *When the Queen dies in London, the Prince of Wales becomes instantaneously king*, when a Ruby/Rails version gets deprecated it also gets instantaneously deprecated on Solidus.

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Summary

We say we're compliant with semver, but we need to define what we understand by
Breaking Change.

Closes #563

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- ~[ ] I have added automated tests to cover my changes.~
- ~[ ] I have attached screenshots to demo visual changes.~
- ~[ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~
- ~[ ] I have updated the README to account for my changes.~

### CODE REVIEW

**Review by elia — APPROVED** [review]
🤩

**Review by gsmendoza — APPROVED** [review]

File: source/release_policy.html.erb [hunk 1]
```diff
@@ -19,8 +19,20 @@ description: Solidus follows semantic versioning and tries to keep a cadence for
...
+        Breaking Change is any update to the Solidus code that can break a host
+        application, except for:
+      </p>
+
+      <ul>
+        <li>Issues due to the application using an interface explicitly marked as private.</li>
+        <li>Issues due to the application monkey patching Solidus code.</li>
+        <li>Issues due to a fix for a security or very important bug for which
+          it hasn't been possible to implement a non-breaking solution.</li>
+        <li>Issues due to the application using a Ruby/Rails version that is no
```
Comment (gsmendoza) [inline source/release_policy.html.erb:33]: @waiting-for-dev Have you considered expanding the scope of this exception to _any_ dependency of Solidus that is no longer maintained?

**Review by gsmendoza — APPROVED** [review]

**Review by kennyadsl — APPROVED** [review]
Nice, thanks @waiting-for-dev!

File: source/release_policy.html.erb (hunk 1 above)
Comment (waiting-for-dev) [reply source/release_policy.html.erb]: Good call. What do you think about the update?

Comment (gsmendoza) [reply source/release_policy.html.erb]: @waiting-for-dev Awesome, thanks!

**Comment by waiting-for-dev** [comment]
CI will be green once we rebase from https://github.com/solidusio/solidus.io/pull/565

**Pull request merged**